src/Uno.Extensions.Core.Generators/PropertySelector/PropertySelectorCandidate.cs

```diff
@@ -1,43 +1,201 @@
 
 	public InvocationExpressionSyntax? Syntax { get; }
 	public FileLinePositionSpan Location { get; }
+	public string? MethodGlobalNamespace { get; }
+	public string? MethodName { get; }
+
+	public EquatableArray<(string key, string accessor)>? Accessors { get; }
+
+	private static string? GenerateAccessor(INamedTypeSymbol selectorType, ArgumentSyntax? selectorArg)
+	{
+		if (selectorArg?.Expression is not SimpleLambdaExpressionSyntax selector)
+		{
+			return null;
+		}
+
+		if (selector.Parameter is null or { IsMissing: true } or { Identifier.ValueText.Length: <= 0 }
+			|| selector.Body is null or { IsMissing: true })
+		{
+			// Delegate is not defined properly yet, we cannot generate.
+			return null;
+		}
+
+		var entityType = selectorType.TypeArguments[0];
+		var propertyType = selectorType.TypeArguments[1];
+
+		var path = PropertySelectorPathResolver.Resolve(selector);
+		var valueAccessor = $@"new {NS.Edition}.ValueAccessor<{entityType}, {propertyType}>(
+			""{path.FullPath}"",
+			{GenerateGetter(path).Align(3)},
+			{GenerateSetter(entityType, path).Align(3)})";
+
+		return valueAccessor.Align(0);
+	}
 
-	public IMethodSymbol? Method { get; }
+	private static string GenerateGetter(PropertySelectorPath path)
+		=> $"entity => entity{path.FullPath}"; // Note: this is expected to be the SimpleLambdaExpressionSyntax selector
+
+	private static string GenerateSetter(ITypeSymbol record, PropertySelectorPath path)
+	{
+		if (path.Parts.Count is 1)
+		{
+			var part = path.Parts[0];
+			var current = OrDefault(record) is { Length: > 0 } orDefault
+				? $"(current {orDefault})"
+				: "current";
+
+			return $"(current, updated_{part.Name}) => {current} with {{ {part.Name} = updated_{part.Name} }}";
+		}
+		else
+		{
+			var count = path.Parts.Count;
+			var type = record;
+			var current = new List<string>(count);
+			var updated = new List<string>(count);
+
+			for (var i = 1; i <= count; i++)
+			{
+				var part = path.Parts[i - 1];
+
+				if (i < count) // 'current_x' is a parameter of the delegate for the leaf element.
+				{
+					type = (type as INamedTypeSymbol)?.FindProperty(part.Name)?.Type;
+					if (type is null)
+					{
+						// The syntax is not valid (trying to get a property which does not exists), we cannot generate yet.
+						// But as anyway the compilation will fail, instead of failing the generator we just generate an exception.
+						return $"(_, __) => throw new InvalidOperationException(\"Cannot resolve property '{part.Name}' on '{type}'\")";
+					}
+
+					current.Add($"var current_{i} = current_{i - 1}{part.Accessor}{OrDefault(type)};");
+				}
+
+				updated.Add($"var updated_{i - 1} = current_{i - 1} with {{ {part.Name} = updated_{i} }};");
+			}
+
+			updated.Reverse();
+			return $@"(current_0, updated_{count}) =>
+			{{
+				{current.Align(4)}
+				{updated.Align(4)}
+
+				return updated_0;
+			}}".Align(0);
+		}
+
+		static string OrDefault(ITypeSymbol type)
```

Comment:
```suggestion
		private static string OrDefault(ITypeSymbol type)
```

Reply:
@dr1rrb It is a local function. Do you want to move it out?

Reply:
Oh my bad, I didn't realized that. No need to move it out ^^